### PR TITLE
ec/curve448: portability fixups.

### PR DESCRIPTION
**Pull request by dot-asm**
Curve448 portability fixups effectively triggered some harmonization changes in other modules.

### CODE REVIEW

**Comment by dot-asm** [comment]
Ah! Red cross from appveyor is relevant, fixup is coming up... But it also exposed warnings that might be viewed as worthy fix...

**Comment by dot-asm** [comment]
Re-confirm is needed [after appveyor is ready]. Strangely enough VC has lesser problems with truncating 32-bit values than 64-bit ones...

**Comment by dot-asm** [comment]
Always wondered what does "dismiss review" does, gonna try it now...

**Comment by dot-asm** [comment]
> VC has lesser problems with truncating 32-bit values than 64-bit ones...

As it turns out former is on next warning level, so that 64-bit values are in fact more valuable to VC than 32-bit ones.

File: crypto/ec/curve25519.c [hunk 1]
```diff
@@ -12,18 +12,17 @@
...
  * Base 2^51 implementation.
  */
 # define BASE_2_51_IMPLEMENTED
 
 typedef uint64_t fe51[5];
 # if !defined(X25519_ASM)
-typedef unsigned __int128 u128;
-# endif
+typedef __uint128_t u128;
+#endif
```
Comment (mattcaswell) [inline crypto/ec/curve25519.c]: Nit: needs a space between "#" and "endif"

Comment (dot-asm) [reply crypto/ec/curve25519.c]: Oh! Will do [with [skip ci]] once appveyor passes...

**Pull request closed without merging**